project-geometry/src/main/java/com/esri/arcgismaps/sample/projectgeometry/MainActivity.kt

```diff
@@ -0,0 +1,158 @@
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    // setup the data binding for the MapView
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    // shows the projection information as a TextView
+    private val infoTextView: TextView by lazy {
+        activityMainBinding.infoTextView
+    }
+
+    // setup the red pin marker image as bitmap drawable
+    private val markerDrawable: BitmapDrawable by lazy {
+        // load the bitmap from resources and create a drawable
+        val bitmap = BitmapFactory.decodeResource(resources, R.drawable.pin_symbol)
+        BitmapDrawable(resources, bitmap)
+    }
+
+    // setup the red pin marker as a Graphic
+    private val markerGraphic: Graphic by lazy {
+        // creates a symbol from the marker drawable
+        Graphic().apply {
+            symbol = PictureMarkerSymbol(markerDrawable).apply {
+                // resize the symbol into a smaller size
+                width = 30f
+                height = 30f
+                // offset in +y axis so the marker spawned
+                // is right on the touch point
+                offsetY = 25f
+            }
+        }
+    }
+
+    // creates a graphic overlay
+    private val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // configure mapView assignments
+        mapView.apply {
+            // create and add a map with a navigation night basemap style
+            map = ArcGISMap(BasemapStyle.ArcGISNavigationNight)
+            // add the graphics overlay to display marker graphics
+            graphicsOverlays.add(graphicsOverlay)
+            // set the default viewpoint to Redlands,CA
+            setViewpoint(Viewpoint(34.058, -117.195, 5e4))
+        }
+
+        lifecycleScope.launch {
+            // check if the map has loaded successfully
+            mapView.map?.load()?.onSuccess {
+                // capture and collect when the user taps on the screen
+                mapView.onSingleTapConfirmed.collect { event ->
+                    event.mapPoint?.let { point -> geoViewTapped(point) }
+                }
+            }?.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map", mapView)
+            }
+        }
+    }
+
+    /**
+     * Handles the SingleTapEvent by drawing a marker and performs a Spatial reference transformation
+     * of the tapped Location using GeometryEngine and displays the result
+     */
+    private fun geoViewTapped(point: Point) {
```

Comment:
Usually in Android programming, event-handling methods like this are prefixed with "on". I suppose it's kind of a legacy thing from when most APIs were callback-based but I think it still helps readability by indicating that this is called somewhere whenever some event happens.
```suggestion
    private fun onGeoViewTapped(point: Point) {
```